client/src/components/UserContainer/index.js

```diff
@@ -0,0 +1,245 @@
+import React, { Component } from 'react';
+import { NavLink } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { SearchOutlined, MenuOutlined, CloseOutlined } from '@ant-design/icons';
+import './style.css';
+import logo from '../../assets/images/logo.png';
+
+class UserContainer extends Component {
+  state = { show: false };
+
+  reverseShow = () => {
+    const { show } = this.state;
+    this.setState({ show: !show });
+  };
+
+  render() {
+    const { show } = this.state;
+    const { rightPageColor, headerLogo, children, isCohortPages } = this.props;
+
+    const rightNavActive =
+      rightPageColor === 'black'
+        ? 'right-nav-active'
+        : 'right-nav-active-white';
+
+    const rightNav =
+      rightPageColor === 'black' ? 'right-nav' : 'right-nav-white';
+
+    return (
+      <div>
+        <div className="main-header">
+          <div className="header-left">
+            <img src={headerLogo} alt="Code Academy" />
+          </div>
+          <div className="header-right">
+            <SearchOutlined style={{ color: rightPageColor }} />
+            <MenuOutlined
+              style={{ color: rightPageColor }}
+              onClick={this.reverseShow}
+            />
+          </div>
+          {show ? (
```

Comment:
Another idea for showing and hiding this is to use css and toggling another classname.  Then your markup is always on the page and the css classname is showing and hiding it using `display: none` or `display: block`.
That could make your component file a little easier to read, since it's one less conditional inside a big render function.  That also makes it very easy for you to add css animation later, to the opening and closing of the menu (you can add a transition and use opacity, or change the height, etc., based on that toggling className).
It's not super important, just an idea for the future!

Reply:
It's a good idea,  thank you 🙏